client/landing/gutenboarding/components/link/index.tsx

```diff
@@ -46,14 +47,16 @@ type Assign3< T extends object, U extends object, V extends object > = Assign< T
 // LinkProps _without_ component
 // LinkProps `to` is optional
 type Props = Omit<
-	Assign3< Button.AnchorProps, Omit< LinkProps, 'component' >, Partial< Pick< LinkProps, 'to' > > >,
+	Assign3< Button.AnchorProps, Omit< LinkProps, 'component' >, { to?: StepType } >,
 	'disabled' | 'href'
 >;
 const Link: FunctionComponent< Props > = ( { children, ...props } ) => {
+	const to = usePath( props.to );
```

Comment:
I don't think the `Link` component should be aware of `usePath`. :thinking: `Link` should continue to be a very general-purpose component (it's a replacement for `react-router`'s own `Link` component, which is basically a thin wrapper around `<a />` or `<button />`). If we `usePath` inside of it, it will suddenly behave differently on routes with language fragments.

Instead, I think it should be the responsibility of the code using `<Link />` to `usePath`, if needed. (The relevant pattern here is separation of concerns :slightly_smiling_face: )

They might seem convenient, but convenience shortcuts like this normally lead to subtle bugs, since developers have a harder time grasping what is done at what stage, and either forget to call it, or use it (unnecessarily) twice. To give one example: In this PR, we are manually calling `usePath` for `<Route />`, but don't need for `<Link />`. I think it'd be preferrable if both had consistent requirements (`usePath` for both, or for neither; with separation of concerns in mind, it should be neither.)

Reply:
You raised a valid point indeed! `<Link>` is now agnostic. The latest commit reflects this now. https://github.com/Automattic/wp-calypso/commit/5e282d4486110b88c2e554619316508a249a2e0